src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                AddList(c.Cmd, "/FI", ForcedIncludeFiles.Split(';'));
+                Add(c.Cmd, "/Yu", PrecompiledHeaderFile);
+
+                Add(c.Cmd, "true".Equals(UndefineAllPreprocessorDefinitions) ? "/u" : "");
+                AddList(c.Cmd, "/D", PreprocessorDefinitions.Split(';'));
+                AddList(c.Cmd, "/U", UndefinePreprocessorDefinitions.Split(';'));
+
+                Add(c.Cmd, ConvertCompileAsAndGetSqLanguage(CompileAs, path, out sqLanguage));
+                Add(c.Cmd, ConvertCompileAsManaged(CompileAsManaged));
+                Add(c.Cmd, "true".Equals(CompileAsWinRT) ? "/ZW" : "");
+                Add(c.Cmd, "true".Equals(DisableLanguageExtensions) ? "/Za" : ""); // defines macro "__STDC__" when compiling C
+                Add(c.Cmd, "false".Equals(TreatWChar_tAsBuiltInType) ? "/Zc:wchar_t-" : ""); // undefines macros "_NATIVE_WCHAR_T_DEFINED" and "_WCHAR_T_DEFINED"
+                Add(c.Cmd, "false".Equals(ForceConformanceInForLoopScope) ? "/Zc:forScope-" : "");
+                Add(c.Cmd, "true".Equals(OpenMPSupport) ? "/openmp" : "");
+
+                Add(c.Cmd, ConvertRuntimeLibrary(RuntimeLibrary));
+                Add(c.Cmd, ConvertExceptionHandling(ExceptionHandling));
+                Add(c.Cmd, ConvertEnableEnhancedInstructionSet(EnableEnhancedInstructionSet));
+                Add(c.Cmd, "true".Equals(OmitDefaultLibName) ? "/Zl" : ""); // defines macro "_VC_NODEFAULTLIB"
+                Add(c.Cmd, "false".Equals(RuntimeTypeInfo) ? "/GR-" : ""); // undefines macro "_CPPRTTI"
+                Add(c.Cmd, ConvertBasicRuntimeChecks(BasicRuntimeChecks));
+
+                // TODO Q: what if it contains space in double quotes?
+                AddList(c.Cmd, AdditionalOptions.Split(' '));
+
+                c.Cmd.Add(AbsoluteFilePath);
+
+                return new Capture[] { p, c };
+            }
+
+            public string ToJson(string path, out string sqLanguage)
+            {
+                Capture[] c = ToCaptures(path, out sqLanguage);
+                string json = "{ \"version\": 0, \"captures\": [ " + c[0].ToJson() + " , " + c[1].ToJson() + " ] }";
+                System.Diagnostics.Debug.WriteLine(json);
+                return json;
+            }
+
+            internal static string ConvertPlatformName(string platformName)
+            {
+                switch (platformName)
+                {
+                    default:
+                        throw new Exception("Unsupported PlatformName: " + platformName);
+                    case "Win32":
+                        return "x86";
+                    case "x64":
+                        return "x64";
+                }
+            }
+
+            internal static string ConvertPlatformToolset(string platformToolset)
+            {
+                switch (platformToolset)
+                {
+                    default:
+                        throw new Exception("Unsupported PlatformToolset: " + platformToolset);
```

Comment:
@Evangelink why in all other similar places you suggest to use `ArgumentException` or `ArgumentOutOfRangeException`, but `UnsupportedPlatformException` here?